src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -306,12 +322,16 @@ inline void rot(Func func, cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<T
             // CUBLAS_POINTER_MODE_DEVICE mode otherwise it causes the segmentation
             // fault. When it is set to device it is users responsibility to
             // synchronise as the function is completely asynchronous.
-            // cublasSetPointerMode(handle, CUBLAS_POINTER_MODE_DEVICE);
+            cublasSetPointerMode(handle, CUBLAS_POINTER_MODE_DEVICE);
```

Comment:
@sbalint98 Sorry for misleading you on the ROT function. Switching to CUBLAS_POINTER_MODE_DEVICE actually causes CUDA_ERROR_ILLEGAL_ADDRESS since there is no scalar result returned by ROT on the device.  

Could you please revert the changes for ROT and also remove the commented out section regarding CUBLAS_POINTER_MODE_DEVICE?

Reply:
Thanks for noticing this, I should have cought it as well.